authz.go

```diff
@@ -121,6 +121,33 @@ type grantStatus struct {
 	via    string                          // the resource check that succeeded for a grant, not set for deny
 }
 
+// useCacheEval - authorize using cache
+func (a *authorizer) useCacheEval(log Logger, principal string, checks []AthenzAccessCheck) *grantStatus {
+	var via string
+	var decision bool
+	var err error
+	for _, check := range checks {
+		decision, err = a.AuthorizationConfig.Config.Cache.authorize(principal, check)
+		if err != nil {
+			log.Println("Error happened using cache to evaluate authorization decision", err)
+		}
+		log.Println("Authorization decision using cache (true=authorized, false=not authorized): ", decision)
```

Comment:
can we move the log to after the loop ends? so, it would reflect the final decision for all checks?